spring-rabbit/src/test/java/org/springframework/amqp/rabbit/AsyncRabbitTemplateTests.java

```diff
@@ -394,6 +396,29 @@ public void testStopCancelled() throws Exception {
 		assertThat(callback.result).isNull();
 	}
 
+	@Test
+	@DirtiesContext
+	public void testConversionException() throws InterruptedException {
+		this.asyncTemplate.getRabbitTemplate().setMessageConverter(new SimpleMessageConverter() {
+			@Override
+			public Object fromMessage(Message message) throws MessageConversionException {
+				throw new MessageConversionException("Failed to convert message");
+			}
+		});
+
+		RabbitConverterFuture<String> replyFuture = this.asyncTemplate.convertSendAndReceive("conversionException");
+
+		final CountDownLatch cdl = new CountDownLatch(1);
+		final AtomicReference<Object> resultRef = new AtomicReference<>();
+        replyFuture.whenComplete((result, ex) -> {
+			resultRef.set(result);
+			cdl.countDown();
+		});
+        assertThat(cdl.await(10, TimeUnit.SECONDS)).isTrue();
+        assertThat(replyFuture).isCompletedExceptionally();
```

Comment:
```suggestion
		assertThat(replyFuture).failsWithin(Duration.ofSeconds(10))
				.withThrowableOfType(MessageConversionException.class)
				.withMessageContaining("Failed to convert message");
```
Something like this?

Reply:
Or never mind. I like it.
You have changed the logic, but at the same it is covered with that `failsWithin()` 😄 
Committing and merging.
Thank you!